sstables/consumer.hh

```diff
@@ -114,8 +116,10 @@ private:
     } _read_int;
 
     // state for READING_BYTES prestate
+    size_t _read_bytes_len = 0;
     temporary_buffer<char> _read_bytes;
-    temporary_buffer<char>* _read_bytes_where; // which temporary_buffer to set, _key or _val?
+    std::vector<temporary_buffer<char>> _read_bytes_fragments;
+    std::variant<temporary_buffer<char>*, fragmented_temporary_buffer*> _read_bytes_where; // which buffer to set, _key, _val, _cell_path or _pk?
```

Comment:
And here, perhaps, you can avoid the variant.

Reply:
Ah, is the variant in order to handle both fragmented keys and fragmented cells?

If so, it's okay as a temporary construct, but let's move quickly towards parsing fragmented keys too.

Reply:
I think we can avoid the variant by having just a `small_vector<temporary_buffer<char>, 1>* _read_bytes_where` and have differently named methods for reading fragmented/non-fragmented values. The variant in question would know whether there are multiple fragments or not.

Reply:
Well, if we move keys towards fragmented buffers then we'll have to merge them again. But yes, the variant is just confusing so better to have two separate methods (and two pre-states etc.). Or bite the bullet and convert keys too.

Reply:
Yet another option is to read keys as a vector<temporary_buffer>, and then when we extract the key convert to a temporary_buffer. Usually the key will be in a single fragment so this is a cheap operation. Later, we can keep keys fragmented all the way.

Reply:
We are already linearizing keys, we just do it as we get the buffers. With my proposal we'd just postpone it. The cost is that we either have to copy the fragment, or keep the buffer containing it alive. Previously we eagerly copied the fragment into the linearizing buffer eagerly.

Reply:
I've removed the variant and merged _read_bytes and _read_bytes_fragments. Now we have separate methods for reading fragmented and non-fragmented values. Notice that if we want to read a non-fragmented value and it's within a single fragment, we return immediately. Otherwise, we still copy fragments into the linearizing buffer, because after keeping them as vector<temporary_buffer> we would need to linearize them anyway.
The linearizing buffer is now front() of the _read_bytes vector.